### PR TITLE
crashes: more tests

### PR DESCRIPTION
**Pull request by matthiaskrgr**
r? @jieyouxu 

try-job: aarch64-apple
try-job: x86_64-msvc

### CODE REVIEW

File: tests/crashes/132765.rs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+//@ known-bug: #123765
```
Comment (jieyouxu) [inline tests/crashes/132765.rs:1]: Problem: wrong issue number 😆

File: tests/crashes/132766.rs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+//@ known-bug: #132766
```
Comment (jieyouxu) [inline tests/crashes/132766.rs:1]: Question:

> This looks like a modified version of tests/ui/inference/ice-ifer-var-leaked-out-of-rollback-122098.rs. It would be nice to indicate that you minimized it from an existing test in the future.

Is that test still present?

File: tests/crashes/133065.rs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+//@ known-bug: #133065
+//@compile-flags: -Zvalidate-mir -Zinline-mir -Zinline-mir-threshold=300
```
Comment (jieyouxu) [inline tests/crashes/133065.rs:2]: Suggestion: hmm, this is also one of those cases where it's better to identify which mir opts are involved, like #133104. Probably split this off into another PR.

File: tests/crashes/133066.rs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+//@ known-bug: #133066
```
Comment (jieyouxu) [inline tests/crashes/133066.rs:1]: Question: can this be further minimized?

File: tests/crashes/133199.rs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+//@ known-bug: #133199
+//@ aux-build: aux133199.rs
```
Comment (jieyouxu) [inline tests/crashes/133199.rs:2]: Discussion: do we want this as a crashes test given we know GCE is incomplete cc @BoxyUwU

**Review by jieyouxu — CHANGES_REQUESTED** [review]
Thanks! I have some questions.

File: tests/crashes/133065.rs (hunk 3 above)
Comment (matthiaskrgr) [reply tests/crashes/133065.rs:2]: as far as I understand this only enables mir inlining.
does mir inlining also enable other opts than mir inlining

File: tests/crashes/133066.rs (hunk 4 above)
Comment (matthiaskrgr) [reply tests/crashes/133066.rs:1]: I spent a couple of minutes on this and didn't manage to produce any meaningful difference :shrug:

File: tests/crashes/133065.rs (hunk 3 above)
Comment (jieyouxu) [reply tests/crashes/133065.rs:2]: cc @saethlin do you know? I'm not super familiar with the MIR-specific ones.

**Comment by jieyouxu** [comment]
Thanks, r=me if the discussions indicate we can go ahead and add the relevant crashes tests.

File: tests/crashes/133199.rs (hunk 5 above)
Comment (matthiaskrgr) [reply tests/crashes/133199.rs:2]: > But yes, every issue should ideally have a test. ICE issues are not special here.

File: tests/crashes/133065.rs (hunk 3 above)
Comment (matthiaskrgr) [reply tests/crashes/133065.rs:2]: I'm not too convinced that this is too useful in general.
If we for example have a broken mir opt in the pipeline, and the mir opt gets banished to -Zunsave-mir-opts, this still fixes the bug from a user perspective as the mir opt is no longer run by default.

Also some mir opts rely on each other which is why I don't fuzz single mir opts, as there are probably bunch of cases where "one opt will break if that other opt is not run before it."

Comment (jieyouxu) [reply tests/crashes/133065.rs:2]: Since this is a crashes test we can also remove it easily if it becomes a problem, so yeah it's a whatever.

**Review by jieyouxu — APPROVED** [review]
LGTM

**Comment by matthiaskrgr** [comment]
I've removed the ones that passed for now, will revisit these later
@bors r=jieyouxu

**Comment by matthiaskrgr** [comment]
@bors r- 

https://github.com/rust-lang/rust/pull/133948#issuecomment-2522486023

**Comment by jieyouxu** [comment]
Maybe arch-specific ABI?

**Review by jieyouxu — APPROVED** [review]
Thanks!

**Comment by jieyouxu** [comment]
#132981 might be arch-specific?

@bors r-
